docs/md/content/usage/tool-authoring.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Authoring Tools"
+description: "Learn how to author a PRM tool."
+category: narrative
+tags:
+  - templates
+  - usage
+weight: 20
+---
+
+## Structure
+
+A PRM tool package can be packaged as a `.tar.gz` or hosted in a Github repository.
+It is comprised of a `prm-config.yml` and an optional `content` directory.
+The `content` directory can contain scripts / binaries / config files, etc... required to supplement the functionality of the tool outside the configurable parameters within the `prm-config.yml`.
+
+A tool can be:
+- A Ruby gem
+- An executable binary
+- A container
+
+## Location
+
+You can specify the location of your tools using the `--toolpath` option:
+
+```bash
+prm exec my-name/my-tool --templatepath /home/me/tools
+```
+
+## Composition
+
+At minimum, a PRM tool must contain a `prm-config.yml` in the root directory.
+An optional `content` directory should also reside at root.
+
+### Required Parameters
+
+Firstly, you'll need to define some metadata about the tool under the `plugin` section:
+
+```yaml
+---
+plugin:
+  author: fantastic-me
+  id: awesome-tool
+  display: Awesome Tool
+  version: 0.1.0
+  upstream_project_url: https://github.com/fantastic-me/awesome-me
+```
+
+To execute the above tool, you would run the command `prm exec fantastic-me/awesome-tool`
+
+### Common Parameters
+
+Next, there is a `common` section that contains a number of parameters with assumed defaults, if not defined:
+
+```
+  Parameter             |  Default Value
+  ──────────────────────|───────────────
+  can_validate          | false
+  help_arg              | '--help'
+  needs_write_access    | false
+  success_exit_code     | 0
+```
+
+`can_validate`: Is this tool designed to validate the target code dir?
+
+`help_arg`: The arg required by the tool to display help info
+
+<!-- Uncomment when this is implemented -->
+<!-- `interleave_stdout_err`: Should the stdout & stderr be interleaved in to one stream, as opposed to separate ones? -->
+
+`needs_write_access`: Will the execution of this tool require RW permissions against the target code dir?
+
+`sucess_exit_code`: What return code should be interpreted as the tool exiting successfully?
+
+
+```yaml
+common:
+  can_validate: true
+  help_arg: '-h'
+  needs_write_access: true
+  success_exit_code: 2
+  use_script: "entrypoint-script"
+```
+
+There are a number of optional parameters that can be defined under `common` too:
+
+`default_args`: A list of args to pass to the invoked tool (this can be overriden with the flag `--toolArgs` flag)
+
+`env`: A list of environment variables and the value to set them to
+
+<!-- Uncomment when this is implemented -->
+<!-- `output_mode`: The output mode (`junit|json|yaml`) -->
+
+`requires_git`: Set to `true` if this tool requires `git`
+
+`use_script`: The name of the script to invoke to execute the tool.
+This should be present in the `content` dir and in either `.sh` or `.ps1` format.
```

Comment:
implementation dependant but this will in `.sh` and `.ps1` format

Reply:
also say that `use_script` should withhold the extension i.e `runme.sh` should be defined `use_script: "runme"`

Reply:
I might drop the reference to the `.sh|ps1` format for now, on second thoughts - especially given my [prior comment](https://github.com/puppetlabs/prm/pull/72/files#r772198442)